wkcuber/tile_cubing.py

```diff
@@ -1,60 +1,204 @@
+            z_number = max(z_number, len(occurrence) - 2)
+    return x_number, y_number, z_number
+
+
+def detect_interval_for_dimensions(
+    file_path_pattern: str,
+    x_decimal_length: int,
+    y_decimal_length: int,
+    z_decimal_length: int,
+) -> Tuple[int, int, int, int, int, int, str, int]:
+    arbitrary_file = None
+    file_count = 0
+    decimal_length = {
+        "x": x_decimal_length,
+        "y": y_decimal_length,
+        "z": z_decimal_length,
+    }
+    current_decimal_length = {"x": 0, "y": 0, "z": 0}
+    max_dimensions = {"x": 0, "y": 0, "z": 0}
+    min_dimensions = {"x": None, "y": None, "z": None}
+
+    # find all files by trying all combinations of dimension lengths
+    for x in range(x_decimal_length + 1):
+        current_decimal_length["x"] = x
+        for y in range(y_decimal_length + 1):
+            current_decimal_length["y"] = y
+            for z in range(z_decimal_length + 1):
+                current_decimal_length["z"] = z
+                specific_pattern = replace_coordinates_with_glob_regex(
+                    file_path_pattern, {"z": z, "y": y, "x": x}
+                )
+                found_files = glob(specific_pattern)
+                file_count += len(found_files)
+                for file in found_files:
+                    occurrences = re.findall("({x+}|{y+}|{z+})", file_path_pattern)
+                    index_offset_caused_by_brackets_and_specific_length = 0
+                    for occurrence in occurrences:
+                        # update the offset since the pattern and the file path have different length
+                        occurrence_begin_index = (
+                            file_path_pattern.index(occurrence)
+                            - index_offset_caused_by_brackets_and_specific_length
+                        )
+                        index_offset_caused_by_brackets_and_specific_length += 2
+                        current_dimension = occurrence[1]
+                        occurrence_end_index = (
+                            occurrence_begin_index
+                            + decimal_length[current_dimension]
+                            - current_decimal_length[current_dimension]
+                        )
+                        index_offset_caused_by_brackets_and_specific_length += current_decimal_length[
+                            current_dimension
+                        ]
+                        coordinate_value = int(
+                            file[occurrence_begin_index:occurrence_end_index]
+                        )
+                        min_dimensions[current_dimension] = min(
+                            min_dimensions[current_dimension] or coordinate_value,
+                            coordinate_value,
+                        )
+                        max_dimensions[current_dimension] = max(
+                            max_dimensions[current_dimension], coordinate_value
+                        )
+
+    return (
```

Comment:
Please add an assertion that verifies that `min_dimensions` does not include None anymore.